src/solve/test.rs

```diff
@@ -175,13 +175,23 @@ fn prove_forall() {
         // Here, we do know that `T: Clone`, so we can.
         goal {
             forall<T> {
-                if (WellFormed(T: Clone), T: Clone) {
+                if (T: Clone) {
                     Vec<T>: Clone
                 }
             }
         } yields {
             "Unique; substitution [], lifetime constraints []"
         }
+
+        goal {
```

Comment:
maybe good to add a comment here that we are explicitly *not* "knowing" that `WF(T: Clone)`, right?